test/middleware/spec.session-model.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const baseController = require('../helpers/controller');
+const sessionModel = require('../../lib/middleware/session-model');
+
+describe('middleware/session-model', () => {
+
+    let BaseController, StubController;
+    let req, res, next, controller;
+
+    beforeEach(() => {
+        req = request();
+        delete req.sessionModel;
+        res = response();
+        next = sinon.stub();
+
+        BaseController = baseController();
+        StubController = sessionModel(BaseController);
+        controller = new StubController({
+            name: 'Wizard-Name'
+        });
+    });
+
+    describe('middlewareSetup override', () => {
+        it('calls the super method', () => {
+            controller.middlewareSetup();
+            BaseController.prototype.middlewareSetup.should.have.been.calledOnce;
+        });
+
+        it('uses the createSessionModel middleware', () => {
+            controller.middlewareSetup();
+            BaseController.prototype.use.should.have.been.calledOnce;
+            BaseController.prototype.use.should.have.been.calledWithExactly(
+                controller.createSessionModel
+            );
+        });
+    });
+
+    describe('createSessionModel', () => {
+        it('throws an error if there is no req.session', () => {
+            delete req.session;
+            expect(() => controller.createSessionModel(req, res, next) ).to.throw();
+        });
+
+        it('creates a new session model on the request and session', () => {
+            controller.createSessionModel(req, res, next);
+            next.should.have.been.calledOnce;
+            next.should.have.been.calledWithExactly();
+            req.sessionModel.should.be.an.object;
+            req.session['hmpo-wizard-Wizard-Name'].should.be.an.object;
+        });
+
+        it('destroys an existing session model before creating a new one', () => {
+            let destroy = sinon.stub();
+            req.sessionModel = {
+                destroy
+            };
+            controller.createSessionModel(req, res, next);
+            destroy.should.have.been.calledOnce;
+        });
+    });
```

Comment:
We also need to test that the next method is called and the wizard model is used.